src/core.js

```diff
@@ -1300,12 +1300,12 @@ $.extend( $.validator, {
 			this.settings.messages[ element.name ].remote = previous.message;
 
 			param = typeof param === "string" && { url: param } || param;
-
-			if ( previous.old === value ) {
+			optionDataString = JSON.stringify( $.param( $.extend( { data: value }, param.data ) ) );
```

Comment:
Yes and $.param converts to the url querystring, which is the 
representation that is sent to the api endpoint. That seems like the more 
correct way of determining whether or not a request was just recently sent 
to the endpoint, no?


On September 8, 2015 4:57:39 AM Markus Staab <notifications@github.com> wrote:

>> @@ -1300,12 +1300,12 @@ $.extend( $.validator, {
>>  			this.settings.messages[ element.name ].remote = previous.message;
>>
>>  			param = typeof param === "string" && { url: param } || param;
>> -
>> -			if ( previous.old === value ) {
>> +			optionDataString = JSON.stringify( $.param( $.extend( { data: value }, 
>> param.data ) ) );
>
> it converts a value into a json string, even objects and arrays:
> https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/jzaefferer/jquery-validation/pull/1570/files#r38903425